gdeep/trainer/tests/test_regularizer.py

```diff
@@ -0,0 +1,117 @@
+from typing import List, Tuple
+
+import torch
+from torch import nn
+from torch.utils.data import Dataset, DataLoader
+from torch.optim import SGD
+import numpy as np
+
+from gdeep.trainer import Trainer
+from gdeep.trainer import RegularizedTrainer
+from gdeep.trainer import TihonovRegularizer
+from gdeep.models import FFNet
+from gdeep.search import GiottoSummaryWriter
+from gdeep.data.datasets import FromArray, DataLoaderBuilder
+from gdeep.search import clean_up_files
+
+from gdeep.utility.custom_types import Tensor
+
+
+class MyDataset(Dataset):
+    def __init__(self):
+        self.x = []
+        for _ in range(100):
+            self.x.append((torch.rand(1, np.random.randint(2, 4))).to(torch.float))
+        self.y = np.array(
+            np.random.randint(2, size=100 * 2).reshape(-1, 2), dtype=np.int64
+        )
+
+    def __len__(self):
+        return 100
+
+    def __getitem__(self, item: int) -> Tuple[Tensor, Tensor]:
+        return self.x[item], self.y[item]
+
+
+def collate_fn(batch_tuple: List):
+    # print(batch_tuple)
+    target = torch.zeros(len(batch_tuple), 4)
+    label = torch.zeros(len(batch_tuple), 2).to(torch.long)
+    for i, batch in enumerate(batch_tuple):
+        source = batch[0]
+        target[i, : len(source[-1])] = source
+        label[i] = torch.tensor(batch[1]).to(torch.long)
+    return target, label
+
+
+# model
+class Model1(nn.Module):
```

Comment:
Give it a more descriptive name like `SmallFullyConnectedNeuralNetwork`

Reply:
this is a verbatim copy of the test_trainer.py, so i left it like that

Reply:
I think it would make sense to add the tests to `test_trainer.py`, then you don't have to copy code.